src/Data/ArrayBuffer/DataView.purs

```diff
@@ -67,113 +50,97 @@ foreign import byteOffset :: DataView -> ByteOffset
-
--- | Fetch int32 value at a certain index in a `DataView`.
-getInt32be :: Getter Int
-getInt32be = getter "getInt32" 4 false
-
-getInt32le :: Getter Int
-getInt32le = getter "getInt32" 4 true
-
--- | Fetch uint8 value at a certain index in a `DataView`.
-getUint8 :: Getter UInt
-getUint8 = getter "getUint8" 1 false
-
--- | Fetch uint16 value at a certain index in a `DataView`.
-getUint16be :: Getter UInt
-getUint16be = getter "getUint16" 2 false
-
-getUint16le :: Getter UInt
-getUint16le = getter "getUint16" 2 true
-
--- | Fetch uint32 value at a certain index in a `DataView`.
-getUint32be :: Getter UInt
-getUint32be = getter "getUint32" 4 false
-
-getUint32le :: Getter UInt
-getUint32le = getter "getUint32" 4 true
-
--- | Fetch float32 value at a certain index in a `DataView`.
-getFloat32be :: Getter Number
-getFloat32be = getter "getFloat32" 4 false
-
-getFloat32le :: Getter Number
-getFloat32le = getter "getFloat32" 4 true
-
--- | Fetch float64 value at a certain index in a `DataView`.
-getFloat64be :: Getter Number
-getFloat64be = getter "getFloat64" 8 false
-
-getFloat64le :: Getter Number
-getFloat64le = getter "getFloat64" 8 true
-
--- | Store int8 value at a certain index in a `DataView`.
-setInt8 :: Setter Int
-setInt8 = setter "setInt8" false
-
--- | Store int16 value at a certain index in a `DataView`.
-setInt16be :: Setter Int
-setInt16be = setter "setInt16" false
-
-setInt16le :: Setter Int
-setInt16le = setter "setInt16" true
-
--- | Store int32 value at a certain index in a `DataView`.
-setInt32be :: Setter Int
-setInt32be = setter "setInt32" false
-
-setInt32le :: Setter Int
-setInt32le = setter "setInt32" true
-
--- | Store uint8 value at a certain index in a `DataView`.
-setUint8 :: Setter UInt
-setUint8 = setter "setUint8" false
-
--- | Store uint16 value at a certain index in a `DataView`.
-setUint16be :: Setter UInt
-setUint16be = setter "setUint16" false
-
-setUint16le :: Setter UInt
-setUint16le = setter "setUint16" true
-
--- | Store uint32 value at a certain index in a `DataView`.
-setUint32be :: Setter UInt
-setUint32be = setter "setUint32" false
-
-setUint32le :: Setter UInt
-setUint32le = setter "setUint32" true
-
--- | Store float32 value at a certain index in a `DataView`.
-setFloat32be :: Setter Number
-setFloat32be = setter "setFloat32" false
-
-setFloat32le :: Setter Number
-setFloat32le = setter "setFloat32" true
-
--- | Store float64 value at a certain index in a `DataView`.
-setFloat64be :: Setter Number
-setFloat64be = setter "setFloat64" false
-
-setFloat64le :: Setter Number
-setFloat64le = setter "setFloat64" true
+data AProxy (a :: ArrayViewType) = AProxy
+
+
+class BinaryValue a t <= DataView (a :: ArrayViewType) t | a -> t where
+  getLE :: AProxy a -> DataView -> ByteOffset -> Effect (Maybe t)
+  getBE :: AProxy a -> DataView -> ByteOffset -> Effect (Maybe t)
+  setBE :: AProxy a -> DataView -> t -> ByteOffset -> Effect Boolean
+  setLE :: AProxy a -> DataView -> t -> ByteOffset -> Effect Boolean
+
```

Comment:
What's this? AFAIU a `DataView` is just a raw chunk of memory that you can access any way you want (peek/poke). It's what you would use to read from a raw buffer (think in binary serialisation protocols).

Reply:
So technically, and this is as far as I understand too, ArrayBuffers are the lowest level interface. DataViews are just as you are saying - byte-indexable array with castable setting and getting - you can caste _how_ a value is set and received (via the `AProxy`), like a 32bit little-endian int or something. TypedArrays are just uniform DataViews for every observed value, and only in big-endian.

Is there something wrong with this binding you're seeing? I'm using the AProxy to help decide how much data to set / get.

Reply:
Looks like I got confused reading the signatures, it's clear now.

I'll work on this again in a couple of days.